### [CVE-2024-8088](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2024-8088)
![](https://img.shields.io/static/v1?label=Product&message=CPython&color=blue)
![](https://img.shields.io/static/v1?label=Version&message=0%20&color=brightgreen)
![](https://img.shields.io/static/v1?label=Version&message=3.10.0%20&color=brightgreen)
![](https://img.shields.io/static/v1?label=Version&message=3.11.0%20&color=brightgreen)
![](https://img.shields.io/static/v1?label=Version&message=3.12.0%20&color=brightgreen)
![](https://img.shields.io/static/v1?label=Version&message=3.13.0a1%20&color=brightgreen)
![](https://img.shields.io/static/v1?label=Version&message=3.9.0%20&color=brightgreen)
![](https://img.shields.io/static/v1?label=Vulnerability&message=CWE-835%20Loop%20with%20Unreachable%20Exit%20Condition%20('Infinite%20Loop')&color=brightgreen)

### Description

There is a HIGH severity vulnerability affecting the CPython "zipfile"module affecting "zipfile.Path". Note that the more common API "zipfile.ZipFile" class is unaffected.When iterating over names of entries in a zip archive (for example, methodsof "zipfile.Path" like "namelist()", "iterdir()", etc)the process can be put into an infinite loop with a maliciously craftedzip archive. This defect applies when reading only metadata or extractingthe contents of the zip archive. Programs that are not handlinguser-controlled zip archives are not affected.

### POC

#### Reference
- https://github.com/python/cpython/commit/795f2597a4be988e2bb19b69ff9958e981cb894e

#### Github
- https://github.com/GitHubForSnap/matrix-commander-gael
- https://github.com/adegoodyer/kubernetes-admin-toolkit

